src/mlpack/methods/neuro_cmaes/neuro_cmaes.hpp

```diff
@@ -0,0 +1,1140 @@
+
+
+template<typename T>
+class CMAES{
+public:
+
+//USER FUNCTIONS TO GET PARAMETER IN VALUES AND ARRAYS
+
+T axisRatio()
+{
+ return maxElement(rgD, params.N) / minElement(rgD, params.N);
+};
+
+T evaluation(){ return countevals; }
+
+T fitness(){ return functionValues[index[0]];}
+
+T fitnessBestEver(){ return xBestEver[params.N];}
+
+T generation(){ return gen;}
+
+T maxEvaluation(){ return params.stopMaxFunEvals;}
+
+T maxIteration(){ return std::ceil(params.stopMaxIter); }
+
+T maxAxisLength(){ return sigma*std::sqrt(maxEW);}
+
+T minAxisLength(){ return sigma*std::sqrt(minEW); }
+
+T maxStdDev(){return sigma*std::sqrt(maxdiagC);}
+
+T minStdDev(){return sigma*std::sqrt(mindiagC);}
+
+T dimension(){return params.N;}
+
+T sampleSize(){return params.lambda;}
+
+T sigmaValue(){return sigma;}
+
+  T* diagonalCovariance()
+  {
+     for(int i = 0; i < params.N; ++i)
+          output[i] = C[i][i];
+        return output;
+  }
+
+  T* diagonalD(){ return rgD; }
+
+  T* standardDeviation()
+  {
+    for(int i = 0; i < params.N; ++i)
+          output[i] = sigma*std::sqrt(C[i][i]);
+        return output;
+  }
+
+ T* XBestEver(){ return xBestEver;}
+
+T* XBest(){return population[index[0]];}
+
+T* XMean(){return xmean;}
+
+private:
+
+  //!< Random number generator.
+  Random<T> rand;
+  //!< CMA-ES parameters.
+  Parameters<T> params;
+
+  //! Step size.
+  T sigma;
+  //! Mean x vector, "parent".
+  T* xmean;
+  //! Best sample ever.
+  T* xBestEver;
+  //! x-vectors, lambda offspring.
+  T** population;
+  //! Sorting index of sample population.
+  int* index;
+  //! History of function values.
+  T* funcValueHistory;
+
+  T chiN;
+  //! Lower triangular matrix: i>=j for C[i][j].
+  T** C;
+  //! Matrix with normalize eigenvectors in columns.
+  T** B;
+  //! Axis lengths.
+  T* rgD;
+
+  //! Anisotropic evolution path (for covariance).
+  T* pc;
+  //! Isotropic evolution path (for step length).
+  T* ps;
+  //! Last mean.
+  T* xold;
+  //! Output vector.
+  T* output;
+  //! B*D*z.
+  T* BDz;
+  //! Temporary (random) vector used in different places.
+  T* tempRandom;
+  //! Objective function values of the population.
+  T* functionValues;
+  //!< Public objective function value array returned by init().
+  T* publicFitness;
+
+  //! Generation number.
+  T gen;
+  //! Algorithm state.
+  enum {INITIALIZED, SAMPLED, UPDATED} state;
+
+  // repeatedly used for output
+  T maxdiagC;
+  T mindiagC;
+  T maxEW;
+  T minEW;
+
+  bool eigensysIsUptodate;
+  bool doCheckEigen; //!< control via signals.par
+  T genOfEigensysUpdate;
+
+  T dMaxSignifKond;
+  T dLastMinEWgroesserNull;
+
+  std::string stopMessage; //!< A message that contains all matched stop criteria.
+
+  /**
+   * Calculating eigenvalues and vectors.
+   * @param rgtmp (input) N+1-dimensional vector for temporal use.
+   * @param diag (output) N eigenvalues.
+   * @param Q (output) Columns are normalized eigenvectors.
+   */
+  void eigen(T* diag, T** Q, T* rgtmp)
```

Comment:
Respect for implementing the eigenvalue decomposition, but why not just use http://arma.sourceforge.net/docs.html#eig_gen or is there something special, that I missed?